org.eclipse.m2e.core.ui/plugin.xml

```diff
@@ -674,4 +686,21 @@
    <extension point="org.eclipse.m2e.core.archetypeCatalogs">
       <remote  url="https://repo1.maven.org/maven2/archetype-catalog.xml" description="Maven Central"/>
    </extension>
+   <extension
+         point="org.eclipse.ui.menus">
+      <menuContribution
+            allPopups="false"
+            locationURI="menu:project">
+            <command
+                  commandId="org.eclipse.m2e.core.ui.command.updateProject"
+                  style="push">
+               <visibleWhen
+                     checkEnabled="false">
+                   <with variable="activeEditorInput">
```

Comment:
Actions in the project menu usually apply to the current selection, not particularly to the editor.

Reply:
I've tried 
```
                       <with variable="selection">
                          <iterate>
                              <adapt
                                    type="org.eclipse.core.resources.IResource">
                                   <test property="org.eclipse.m2e.hasMavenNature" value="true"/>
                              </adapt>
                          </iterate>
                       </with>
```
but in that case, the menu item disappeared very easily. I mean, even if a java class is selected in the package explorer, I just had to edit that file, it goes away. Maybe, because that 'selection' turns into a text-selection, or something like that. I couldn't figure out the exact details, unfortunately.
What about having the 2 conditions, just or-ed together? It works better in my experience